pkg/cloud/aws/services/kubeadm/aws_defaults.go

```diff
@@ -193,7 +193,11 @@ func SetJoinNodeConfigurationOverrides(caCertHash, bootstrapToken string, machin
 	}
 	out.NodeRegistration.KubeletExtraArgs["cloud-provider"] = CloudProvider
 	if !util.IsControlPlaneMachine(machine.GetMachine()) {
-		out.NodeRegistration.KubeletExtraArgs["node-labels"] = nodeRole
+		if labels, ok := out.NodeRegistration.KubeletExtraArgs["node-labels"]; ok {
+			out.NodeRegistration.KubeletExtraArgs["node-labels"] = fmt.Sprintf("%s,%s", labels, nodeRole)
```

Comment:
Should we use strings.Split and strings.Join instead of manual concatenation?

Reply:
Could we add a test case as well?

Reply:
Yep I agree with this. I believe I have done all the things. 😸